packages/arkor-local/python/mlx/train_shim.py

```
@@ -0,0 +1,306 @@
+    elif warmup > 0:
+        # Constant schedule with warmup: a flat "linear" segment after the
+        # warmup join keeps the effective rate constant.
+        config = {"name": "linear_schedule", "arguments": [lr, lr, max(1, iters)]}
+    else:
+        return None
+    if warmup > 0:
+        config["warmup"] = int(warmup)
+    return config
+
+
+def _make_callback(run, prepared, iters, batch_size, raw_dir: Path, adapters_dir: Path):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_make_callback`

(ANN202)